### ISSUE TITLE
npm install errors need some context

### ISSUE THREAD

**Issue by danielo515** [lerna/lerna#952]
Hello,

We do know that lerna swallows error messages a bit as stated on #119 
One important bug is when certain package does not exist on the registry. It is tricky to see from which package is the error coming, and if the remote repository does not provides the package name, then you are totally busted. This error occurs frequently when you mistype a local package in any of the other packages, then lerna will try to install it from npm and it will fail.

## Expected Behavior
I want to see on a line saying which package is going to be installed which package, or at least on which subpackage installation is being performed.

## Current Behavior
I can only see a subset of the npm log in the lerna-debug.log, which is not sufficient to see where the error is.

## Possible Solution
Lerna should provide some context about what package is being processed (where the npm install is being executed, for example) and to provide the full npm output, and not just a part of it.

## Steps to Reproduce (for bugs)
Just require a non existing package in any sub-package.

<details><summary>lerna.json</summary><p>

```json
{
  "lerna": "2.0.0",
  "packages": [
    ".",
    "src/*"
  ],
  "hoist": true,
  "registry": "https://tools.caseonit.com/npm",
  "commands": {
    "publish": {
      "conventionalCommits": true,
      "message": "chore: Publish"
    }
  },
  "version
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
478 silly getExecOpts      npm_package_scripts_precommit: 'validate-commit-msg',
478 silly getExecOpts      _system_name: 'OSX',
478 silly getExecOpts      npm_node_execpath: '/usr/local/Cellar/node/7.7.1/bin/node',
478 silly getExecOpts      npm_config_link: '',
478 silly getExecOpts      npm_config_prefix: '/usr/local' } }
479 silly installInDir [ 'npm', [ 'install' ] ]
480 silly installInDir cleanup /Users/danielo/GIT/case-alert/package.json
481 silly renameSync [ '/Users/danielo/GIT/case-alert/package.json.lerna_backup',
481 silly renameSync   '/Users/danielo/GIT/case-alert/package.json' ]
482 error execute Error: Command failed: npm install
482 error execute npm ERR! Darwin 16.6.0
482 error execute npm ERR! argv "/usr/local/Cellar/node/7.7.1/bin/node" "/usr/local/bin/npm" "install"
482 error execute npm ERR! node v7.7.1
482 error execute npm ERR! npm  v4.1.2
482 error execute npm ERR! code E404
482 error execute
482 error execute npm ERR! 404 no such package available : npm
482 error execute npm ERR! 404
482 error execute npm ERR! 404  'npm' is not in the npm registry.
482 error execute npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
482 error execute npm ERR! 404
482 error execute npm ERR! 404 Note that you can also install from a
482 error execute npm ERR! 404 tarball, folder, http url, or git url.
482 error execute
482 error execute npm ERR! Please include the following file with any support request:
482 error execute npm ERR!     /Users/danielo/GIT/case-alert/npm-debug.log
482 error execute
482 error execute     at Promise.all.then.arr (/Users/danielo/GIT/case-alert/node_modules/execa/index.js:210:11)
482 error execute     at process._tickCallback (internal/process/next_tick.js:109:7)
482 error execute  callback with error
482 error execute  { Error: Command failed: npm install
482 error execute npm ERR! Darwin 16.6.0
482 error execute npm ERR! argv "/usr/local/Cellar/node/7.7.1/bin/node" "/usr/local/bin/npm" "install"
482 error execute npm ERR! node v7.7.1
482 error execute npm ERR! npm  v4.1.2
482 error execute npm ERR! code E404
482 error execute
482 error execute npm ERR! 404 no such package available : npm
482 error execute npm ERR! 404
482 error execute npm ERR! 404  'npm' is not in the npm registry.
482 error execute npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
482 error execute npm ERR! 404
482 error execute npm ERR! 404 Note that you can also install from a
482 error execute npm ERR! 404 tarball, folder, http url, or git url.
482 error execute
482 error execute npm ERR! Please include the following file with any support request:
482 error execute npm ERR!     /Users/danielo/GIT/case-alert/npm-debug.log
482 error execute
482 error execute     at Promise.all.then.arr (/Users/danielo/GIT/case-alert/node_modules/execa/index.js:210:11)
482 error execute     at process._tickCallback (internal/process/next_tick.js:109:7)
482 error execute   stack: 'Error: Command failed: npm install\nnpm ERR! Darwin 16.6.0\nnpm ERR! argv "/usr/local/Cellar/node/7.7.1/bin/node" "/usr/local/bin/npm" "install"\nnpm ERR! node v7.7.1\nnpm ERR! npm  v4.1.2\nnpm ERR! code E404\n\nnpm ERR! 404 no such package available : npm\nnpm ERR! 404 \nnpm ERR! 404  \'npm\' is not in the npm registry.\nnpm ERR! 404 You should bug the author to publish it (or use the name yourself!)\nnpm ERR! 404 \nnpm ERR! 404 Note that you can also install from a\nnpm ERR! 404 tarball, folder, http url, or git url.\n\nnpm ERR! Please include the following file with any support request:\nnpm ERR!     /Users/danielo/GIT/case-alert/npm-debug.log\n\n    at Promise.all.then.arr (/Users/danielo/GIT/case-alert/node_modules/execa/index.js:210:11)\n    at process._tickCallback (internal/process/next_tick.js:109:7)',
482 error execute   code: 1,
482 error execute   killed: false,
482 error execute   stdout: '',
482 error execute   stderr: 'npm ERR! Darwin 16.6.0\nnpm ERR! argv "/usr/local/Cellar/node/7.7.1/bin/node" "/usr/local/bin/npm" "install"\nnpm ERR! node v7.7.1\nnpm ERR! npm  v4.1.2\nnpm ERR! code E404\n\nnpm ERR! 404 no such package available : npm\nnpm ERR! 404 \nnpm ERR! 404  \'npm\' is not in the npm registry.\nnpm ERR! 404 You should bug the author to publish it (or use the name yourself!)\nnpm ERR! 404 \nnpm ERR! 404 Note that you can also install from a\nnpm ERR! 404 tarball, folder, http url, or git url.\n\nnpm ERR! Please include the following file with any support request:\nnpm ERR!     /Users/danielo/GIT/case-alert/npm-debug.log\n',
482 error execute   failed: true,
482 error execute   signal: null,
482 error execute   cmd: 'npm install',
482 error execute   timedOut: false }

```
</p></details>

**Comment by evocateur** [comment]
#1004 is working on this

**Comment by iabw** [comment]
#1004 was released in Lerna 2.2.0, but I'm on Lerna 2.4.0 and still want to +1 this. Here's an install error I was getting earlier:

```
👻  $ lerna bootstrap
lerna info version 2.4.0
lerna info Bootstrapping 7 packages
lerna info lifecycle preinstall
lerna info Installing external dependencies
lerna ERR! execute callback with error
lerna ERR! Error: Command failed: npm install
lerna ERR! npm WARN deprecated babel-preset-es2015@6.24.1: 🙌  Thanks for using Babel: we recommend using babel-preset-env now: please read babeljs.io/env to update!
lerna ERR! npm ERR! code ENOLOCAL
lerna ERR! npm ERR! Could not install from "node_modules/prop-types" as it does not contain a package.json file.
lerna ERR!
lerna ERR! npm ERR! A complete log of this run can be found in:
lerna ERR! npm ERR!     /Users/iabw/.npm/_logs/2017-10-28T23_05_02_344Z-debug.log
lerna ERR!
lerna ERR!     at Promise.all.then.arr (/usr/local/lib/node_modules/lerna/node_modules/execa/index.js:236:11)
lerna ERR!     at <anonymous>
(node:25282) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 4): Error: Command failed: npm install
npm WARN deprecated babel-preset-es2015@6.24.1: 🙌  Thanks for using Babel: we recommend using babel-preset-env now: please read babeljs.io/env to update!
npm ERR! code ENOLOCAL
npm ERR! Could not install from "node_modules/prop-types" as it does not contain a package.json file.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/iabw/.npm/_logs/2017-10-28T23_05_02_344Z-debug.log

(node:25282) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

The full log file referenced the path, so I used that to find the correct package, but it's not ideal.

PS - basically unrelated, but also annoying/weird - the prop-types bug was caused by Lerna apparently moving some peerDependencies to dependencies, and the "new" prop-types dependency was referenced like "node_modules/prop-types", (which I realized to check after googling and landing on this [ionic-cli issue](https://github.com/ionic-team/ionic-cli/issues/2454#issuecomment-320839317)). These are pretty basic packages, just small React/Babel/Webpacked components, and I'm not using ionic-cli or Cordova.
